Documentation/ceph-upgrade.md

```diff
@@ -89,6 +89,40 @@ you should change to match where your images are located.
         value: "quay.io/k8scsi/csi-resizer:v0.4.0"
 ```
 
+### Retroactively enabling volume expansion
+
+When upgrading to the CSI 2.0 driver with existing persistent volumes, the following considerations should be taken into account when choosing to leverage the volume expansion feature:
+
+#### Remediate existing storage classes
+
+It is also necessary to ensure that the controlling StorageClass have the required changes
+to support Volume Expansion see the documentation about configuring [dynamically expanding volumes](https://rook.io/docs/rook/master/ceph-csi-drivers.html#dynamically-expand-volume).
+
+#### Remediate existing persistent volumes
+
+For any existing persistent volumes that you wish to 'upgrade' to allow expansion, you must
+patch them with some needed metadata.
+
+You can do this by:
+
+1. create a patch file, e.g. `csi-expansion-patch.yaml` with the following:
+
+```yaml
+spec:
+  csi:
+    controllerExpandSecretRef:
+      name: rook-ceph-csi
+      namespace: rook-ceph
+```
+
+(replace the secret name and namespace if you are doing something different than default)
+
+2. Patch each pv with something like,
+
+```yaml
+kubectl patch pv $(kubectl get pvc <some pvc> -o=jsonpath='{.spec.volumeName}') --patch "$(cat csi-expansion-patch.yaml)"
```

Comment:
Is there anything in the K8s CSI docs that indicate how to patch a PV like this? It sure would be nice to have a cleaner way to update the PV. I can't think of a better way though.

Reply:
Not that I could find.  The instructions could be simplified to be just something like `kubectl patch pv <some pv> --patch "$(cat csi-expansion-patch.yaml)` and identifying which PV to patch could be an exercise left up to the user.

I also explored some way to do an inline patch to add the necessary `spec.csi.controllerExpandSecretRef...` part but never found a way to make it work.  Doing a patch inline would be better than making an intermediate patch file IMO.

We could also consider updating the instructions with just verbiage instructing the users to 'edit' the PV with the following block and how to go about doing that could be an exercise left to them?